scipy/ndimage/tests/test_ndimage.py

```diff
@@ -4578,3 +4578,16 @@ def test_dilation_scalar_size(self):
         result = ndimage.grey_dilation(self.array, size=3)
         assert_array_almost_equal(result, self.dilated3x3)
 
+class TestBinaryOpening:
+
+    def setup_method(self):
+        a = numpy.zeros((5,5), dtype=int)
```

Comment:
Also, an array of all zeros is not modified by opening or closing: it may be a better choice to use a `np.ones` array here, although it doesn't really matter much...

Reply:
The next two lines modify the array `a`. The array assigned to `self.array` has both ones and zeros.